### ISSUE TITLE
Background Fetch API throws for extension protocol

### ISSUE THREAD

**Issue by guest271314** [mdn/content#12370]
MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Background_Fetch_API

#### What information was incorrect, unhelpful, or incomplete?

Neither the README.md https://github.com/WICG/background-fetch#readme, specification https://wicg.github.io/background-fetch/, nor MDN documentation explains that background fetch will throw for browser extension protocol (not HTTPS or HTTP).

#### Specific section or headline?

None.

#### What did you expect to see?

"only the https: scheme is allowed, or http: for loopback IPs."

#### Did you test this? If so, how?

Yes.

Run the following code in an extension

```
try {
    const registration = await navigator.serviceWorker.ready;
    console.log(registration);
    const bgFetchReg = await registration.backgroundFetch.fetch('test', ['./worker.js']);
  } catch(err) {
    console.error(err);
  }
```

observe the `TypeError` thrown

```
TypeError: Failed to execute 'fetch' on 'BackgroundFetchManager': Refused to fetch 'chrome-extension://xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx/stream' because only the https: scheme is allowed, or http: for loopback IPs.
```

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/api/background_fetch_api`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Background_Fetch_API
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/background_fetch_api/index.md
* Last commit: https://github.com/mdn/content/commit/2279e5ae6c229c707a014a22aa1ec4635a0f981f
* Document last modified: 2021-09-14T18:08:09.000Z

</details>

**Comment by guest271314** [comment]
No. This appears to be an omission. The `BackgroundFetch` specification authors were not thinking about MV3 `ServiceWorker` usage, and MV3 authors developing background `ServiceWorker` were not thinking about `BackgroundFetch` - even within the same organization there was a disconnect, not a unified collaborative approach, apparently, hence this one of several omissions, TL;DR this case can be added to the list at https://github.com/w3c/webextensions/issues/72. 

People and organizations make mistakes. Whether they admit those mistakes or not is a different issue. Empirical evidence outweighs denials.

Chromium/Chrome Manifest Version 3 `ServiceWorker` https://developer.chrome.com/docs/extensions/mv3/migrating_to_service_workers/ can `fetch()` resources across protocols. To prove that I will attach a screenshot of fetching a chrome-extension: protocol resource from this https: GitHib page, listed in the extensions' `"web_accessible_resources"` field. I used the `BackgroundFetch` code within the extension itself, the same protocol.

![fetch_chrome_extension_protocol_from_https_protocol](https://user-images.githubusercontent.com/4174848/150794854-70e7904c-c23b-4cd6-9b79-7a22468034d4.png)

**Comment by sideshowbarker** [comment]
alerting @jakearchibald

**Comment by jakearchibald** [comment]
`BackgroundFetch` does not keep the service worker alive. In fact, it's whole reason for existing is to avoid keeping the service worker alive.

The service worker is not supposed to stay alive, and attempts to make it stay alive will be buggy.

Shared workers however do stay alive.

If you think extensions should be able to use shared workers, that's something to propose to extension folks.

**Comment by guest271314** [comment]
I can already keep `ServiceWorker` active indefintely using a variety of approaches, not all listed in the above repository.

I never ending `fetch()` request will keep the `ServiceWorker` active indefinitely. My use case just happens to be streaming media.

You folks need to document that `BackgroundFetch` is restricted for chrome-extension: and say why. Not documented anywhere.

**Comment by rebloor** [comment]
I'm closing this issue because:

- there's been no activity for over three years
- the code snippet doesn't appear to run in an extension
- from the discussion, it's unclear that a change is needed
